pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -130,6 +146,16 @@ def sum(input: Tensor, dim: Optional[int] = None, dtype: Optional[Union[str, _dt
   return modules.Reduce(mode="sum", axes=dim)(input)
 
 
+def max(input: Tensor, dim: Optional[int] = None, dtype: Optional[Union[str, _dtype]] = None) -> Tensor:
+  out = input
+  if dim is None:
```

Comment:
1. You should not make this a separate case here but let `Reduce` handle this logic.

2. Btw, in `sum(...)` here, you should remove the `assert dim is not None`. `Reduce` should handle that, or if this is not implemented yet, then `Reduce` should fail.

3. Also, I just noticed that `Reduce` is implemented incorrectly. It should not accept `str` for `axes`. Instead though, it should also accept `None` or a `List[int]`. And it must use `_get_input_axis_to_returnn` in `Reduce.create_returnn_layer_dict` for the axes.

Reply:
Ok, I fixed this now.